examples/gitlab/example.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2024 OpenPubkey
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package gitlab_example
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/openpubkey/openpubkey/client"
+	"github.com/openpubkey/openpubkey/client/providers"
+	"github.com/openpubkey/openpubkey/verifier"
+)
+
+func SignWithGitlab() error {
+
+	op, err := providers.NewGitlabOpFromEnvironment("OPENPUBKEY_JWT")
+	if err != nil {
+		return err
+	}
+	opkClient, err := client.New(op)
+	if err != nil {
+		return err
+	}
+
```

Comment:
for examples, its always nice to have lots of comments describing each step

Reply:
Good point, added some comments